app/models/user.rb

```diff
@@ -0,0 +1,35 @@
+class User < ActiveRecord::Base
+  has_many :recipes, dependent: :destroy
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
+  validates :email, presence: true,
+                    format: { with: VALID_EMAIL_REGEX },
+                    uniqueness: true
+  has_secure_password
+
+  before_create :generate_auth_token
+  after_create :generate_recipes
+
+private
+  def generate_auth_token
+    begin
+      generated_token = SecureRandom.hex(10)
+    end while User.where(auth_token: generated_token).exists?
+
+    self.auth_token = generated_token
+  end
+
+  def generate_recipes
+    if self.seed_recipes == true
+      recipes = YAML::load(File.open("#{Rails.root}/config/recipies.yml"))
+      recipes.each do |recipe|
+        self.recipes.create(
+          name: recipe["recipe"]["name"],
+          difficulty: recipe["recipe"]["difficulty"],
+          instructions: recipe["recipe"]["instructions"],
+          favorite: recipe["recipe"]["favorite"],
+          remote_photo_url: recipe["recipe"]["remote_photo_url"],
+          description: recipe["recipe"]["description"])
+      end
+    end
+  end
```

Comment:
@jgorset I moved the method's logic into a class, thanks for the suggestion it looks much better now, you can see the commit [there](https://github.com/karimmtarek/hyper-recipes/commit/42d7d276c06353c8e749a239b27e05facc242ef4)

**Two thing I'm not sure about:**
1. Is it right to host the new class in models directory? if not where should it go?
2. Where should the recipes yml file be hosted? is it okay having it in the config directory?